devito/ops/types.py

```diff
@@ -147,3 +147,18 @@ def __init__(self, name, *args, **kwargs):
     @property
     def _C_typename(self):
         return namespace['ops_dat_type']
+
+
+class TypeCast(basic.Object):
+    """
+        A node encapsulating a cast to a raw C type.
+    """
+
+    def __init__(self, name, dtype, cast_to_ctype):
+        self.cast_to_ctype = cast_to_ctype
+        self.dtype = dtype
+        self.name = str(name)
```

Comment:
Error during apply():
```
File "devito/types/basic.py", line 974, in _arg_defaults
    if callable(self.value):
AttributeError: 'TypeCast' object has no attribute 'value'
```

Reply:
I think it's time to add a test for a basic apply in our `test_ops.py`.

I'll try to add a simple time marching test in this PR. :crossed_fingers:

Reply:
The CI checks will fail with an apply test. How do you intend to manage that?

Reply:
Right, so no testing this for now.

Reply:
Fixed with the last commit